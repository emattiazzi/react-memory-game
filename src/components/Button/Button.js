import React from "react";
import PropTypes from "prop-types";

const styles = {
  display: "inline-block",
  backgroundColor: "#388BCB",
  border: "2px solid #388BCB",
  fontSize: "0.875rem",
  lineHeight: "2rem",
  color: "#FFFFFF",
  padding: "0.25rem 0.5rem",
  borderRadius: "0.25rem",
  outline: "0",
  cursor: "pointer"
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default function Button({ label, onClick }) {
  return (
    <button onClick={onClick} style={styles}>
      {label}
    </button>
  );
}
