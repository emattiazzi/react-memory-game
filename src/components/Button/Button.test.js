import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './Button';
configure({ adapter: new Adapter() });

describe('<Button />', () => {
  it('should render a Button with label and onClick props', () => {
    const buttonLabel = 'A button';
    const wrapper = shallow(<Button onClick={() => {}} label={buttonLabel} />);
    expect(wrapper.text()).toBe(buttonLabel);
  })

  it('matches the snapshot', () => {
    const tree = shallow(<Button onClick={() => {}} label="Snapshot" />);
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('calls onClick when clicked ', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick} label="Simulate click" />);
    const button = wrapper.find('button');

    button.simulate('click');
    expect(onClick).toHaveBeenCalled();
  })
})