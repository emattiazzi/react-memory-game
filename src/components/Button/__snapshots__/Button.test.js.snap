// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches the snapshot 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#388BCB",
      "border": "2px solid #388BCB",
      "borderRadius": "0.25rem",
      "color": "#FFFFFF",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "0.875rem",
      "lineHeight": "2rem",
      "outline": "0",
      "padding": "0.25rem 0.5rem",
    }
  }
>
  Snapshot
</button>
`;
